app/move/controllers/unassign.js

```diff
@@ -0,0 +1,64 @@
+const FormWizardController = require('../../../common/controllers/form-wizard')
+const allocationService = require('../../../common/services/allocation')
+const moveService = require('../../../common/services/move')
+
+class UnassignController extends FormWizardController {
+  middlewareChecks() {
+    this.use(this.checkAllocation)
+    super.middlewareChecks()
+  }
+
+  async checkAllocation(req, res, next) {
+    const { move } = res.locals
+
+    if (!move.allocation) {
+      return res.redirect(`/move/${move.id}`)
+    }
+    if (!move.person) {
+      return res.redirect(`/allocation/${move.allocation.id}`)
+    }
+
+    next()
+  }
+
+  middlewareLocals() {
+    super.middlewareLocals()
+    this.use(this.setMoveRelationships)
+  }
+
+  async setMoveRelationships(req, res, next) {
+    try {
+      const { move } = res.locals
+
+      const allocation = await allocationService.getById(move.allocation.id)
```

Comment:
Taking another look I don't think we use the rest of the details in this journey so we can remove the need to make this service call and also remove the `async` and `try/catch` from the function.

Reply:
👍